src/Compilers/Core/Portable/Operations/IRecursivePatternOperation.cs

```diff
@@ -13,6 +13,25 @@ namespace Microsoft.CodeAnalysis.Operations
     /// </remarks>
     public interface IRecursivePatternOperation : IPatternOperation
     {
+        /// <summary>
+        /// The (explicit or implicit) type accepted for the recursive pattern.
+        /// </summary>
+        ITypeSymbol MatchedType { get; }
+        /// <summary>
+        /// The Deconstruct symbol, if any, used for the deconstruction subpatterns.
+        /// </summary>
+        ISymbol DeconstructSymbol { get; }
+        /// <summary>
+        /// If there is a deconstruction or positional subpattern, this contains the patterns contained within it.
+        /// If there is no deconstruction subpattern, this is a default immutable array.
```

Comment:
>default immutable array [](start = 64, length = 23)

We do not use default in IOperation. This needs to be the empty array.

Reply:
Not sure how this would work for `if (x is ()) { ... }` which matches a 0-element tuple.  In this case we have no matched type, deconstruct symbol, or patterns.  How does the consumer of this API know we are checking for a 0-element tuple type?  Perhaps because that is the only thing it could mean in correct code?

---
In reply to: [243437496](https://github.com/dotnet/roslyn/pull/31967#discussion_r243437496) [](ancestors = 243437496,243430550)